cmd/cluster-agent/app/app.go

```diff
@@ -416,3 +433,23 @@ func setupClusterCheck(ctx context.Context) (*clusterchecks.Handler, error) {
 	log.Info("Started cluster check Autodiscovery")
 	return handler, nil
 }
+
+func initializeRemoteConfig(ctx context.Context) (*remote.Client, error) {
+	configService, err := remoteconfig.NewService()
+	if err != nil {
+		return nil, fmt.Errorf("unable to create remote-config service: %v", err)
+	}
+
+	if err := configService.Start(ctx); err != nil {
+		return nil, fmt.Errorf("unable to start remote-config service: %v", err)
+	}
+
+	rcClient, err := remote.NewClient("cluster-agent", configService, version.AgentVersion, []data.Product{}, time.Second*5)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create local remote-config client: %v", err)
+	}
+
+	rcClient.Start()
```

Comment:
Nitpick: I'd rather run `rcClient.Start()` in the `start` (caller) function since `initializeRemoteConfig` returns the client anyway, and call `defer rcClient.Close()` right after it.

Reply:
That works for me!

The client is primarily returned for product code to register its callbacks on eventually - there aren't any in my PR but your changes that build on this will need the handle to the client to do that. That being said I like making the Start() more deliberate and not hidden in the initialize function.